proc/api.thrift

```diff
@@ -0,0 +1,40 @@
+service Proc {
+  
+  exception InvalidRequest {
+    1: optional string message
+  }
+
+  exception ScootTooBusy {
+    1: optional i64 retryAfterMs
+  }
+  
+  enum JobType {
+    IRON_TESTS=1,
+    DEFAULT=2,
+  }
+
+  struct Command {
+    1: list<string> cmds
+  }
+
+  struct Task {
+    1: optional string id
+    2: required Command command,
+    3: optional string snapshotId,
+    4: optional string contextId
+  }
+
+  struct Job {
+    1: required string id,
+    2: required list<Task> tasks
+  }
+
+  Job RunJob(
+    1: optional string contextId,
+    2: optional JobType jobType,
+    3: required list<Task> tasks
```

Comment:
Required is hard to upgrade, changing it is a breaking change, but having no required fields is a little weird, it suggests your api is endlessly malleable, and can make progress without any real input.  

I chose to make the things that our API will not function without required.  JobId is core to the ScootAPI if we didn't send you one back then you could never check the status of your event.  Likewise a Job without tasks doesn't make sense, a Job needs to specify work to do.  If any of these things changed I would not modify the existing API I would just make a new one since likely the semantics of the API have drastically changed.  

Does that make sense?  Also it means less error handling boiler plate code for us to write, to check for things we always expect to be there